In [6]:

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import ast

import warnings
warnings.filterwarnings('ignore')



In [7]:
zomatodt = pd.read_csv("D:\Data Trained - Excel links\zomato.csv", encoding = "ISO-8859-1")

In [8]:
# Viewing shape of the data
zomatodt.shape

(9551, 21)

In [9]:
# Viewing the top 3 rows of the dataset
zomatodt.head(3)

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  

[3 rows x 21 columns]

From the above data, we can observe the following:

Restaurant Data: 
Location Data: 
Food Data: 
Costing Data: 
Order Details: 
Restaurant Rating Details: 

we proceed to understand the data, summary statistics, data types and missing values better.

In [10]:
# Reading country code 
countrydt = pd.read_excel("D:\Data Trained - Excel links\Country-Code.xlsx")

In [11]:
# Viewing the shape of the dataset
countrydt.shape

(15, 2)

In [12]:
# Viewing the top 5 results of the country Dataframe
countrydt.head()

Country Code    Country
0             1      India
1            14  Australia
2            30     Brazil
3            37     Canada
4            94  Indonesia

In [13]:
# Viewing the Zomato Dataset
zomatodt.head(3)

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   

                           Cuisines  ...          Currency Has Table booking  \
0        French, Japanese, Desserts  ...  Botswana Pula(P)               Yes   
1                          Japanese  ...  Botswana Pula(P)               Yes   
2  Seafood, Asian, Filipino, Indian  ...  Botswana Pula(P)               Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   

   Aggregate rating  Rating color Rating text Votes  
0               4.8    Dark Green   Excellent   314  
1               4.5    Dark Green   Excellent   591  
2               4.4         Green   Very Good   270  

[3 rows x 21 columns]

Now Merging the Zomato and Country Datasets

In [14]:
# Merging the zomato dataset at Restaurant Id level and Country data at country code level
zomatoCountrydt = pd.merge(zomatodt, countrydt, on = 'Country Code') 

In [15]:
# Viewing the head of the 
zomatoCountrydt.head(3)

Restaurant ID         Restaurant Name  Country Code              City  \
0        6317637        Le Petit Souffle           162       Makati City   
1        6304287        Izakaya Kikufuji           162       Makati City   
2        6300002  Heat - Edsa Shangri-La           162  Mandaluyong City   

                                             Address  \
0  Third Floor, Century City Mall, Kalayaan Avenu...   
1  Little Tokyo, 2277 Chino Roces Avenue, Legaspi...   
2  Edsa Shangri-La, 1 Garden Way, Ortigas, Mandal...   

                                     Locality  \
0   Century City Mall, Poblacion, Makati City   
1  Little Tokyo, Legaspi Village, Makati City   
2  Edsa Shangri-La, Ortigas, Mandaluyong City   

                                    Locality Verbose   Longitude   Latitude  \
0  Century City Mall, Poblacion, Makati City, Mak...  121.027535  14.565443   
1  Little Tokyo, Legaspi Village, Makati City, Ma...  121.014101  14.553708   
2  Edsa Shangri-La, Ortigas, Mandaluyong City, Ma...  121.056831  14.581404   

                           Cuisines  ...  Has Table booking  \
0        French, Japanese, Desserts  ...                Yes   
1                          Japanese  ...                Yes   
2  Seafood, Asian, Filipino, Indian  ...                Yes   

  Has Online delivery Is delivering now Switch to order menu Price range  \
0                  No                No                   No           3   
1                  No                No                   No           3   
2                  No                No                   No           4   

  Aggregate rating  Rating color  Rating text Votes      Country  
0              4.8    Dark Green    Excellent   314  Phillipines  
1              4.5    Dark Green    Excellent   591  Phillipines  
2              4.4         Green    Very Good   270  Phillipines  

[3 rows x 22 columns]

In [16]:
# Summary of the dataset now
print(zomatoCountrydt.info(verbose=True))

<class 'pandas.core.frame.DataFrame'>
Int64Index: 9551 entries, 0 to 9550
Data columns (total 22 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Restaurant ID         9551 non-null   int64  
 1   Restaurant Name       9551 non-null   object 
 2   Country Code          9551 non-null   int64  
 3   City                  9551 non-null   object 
 4   Address               9551 non-null   object 
 5   Locality              9551 non-null   object 
 6   Locality Verbose      9551 non-null   object 
 7   Longitude             9551 non-null   float64
 8   Latitude              9551 non-null   float64
 9   Cuisines              9542 non-null   object 
 10  Average Cost for two  9551 non-null   int64  
 11  Currency              9551 non-null   object 
 12  Has Table booking     9551 non-null   object 
 13  Has Online delivery   9551 non-null   object 
 14  Is delivering now     9551 non-null   object 
 15  Switch to order menu 

MISSING VALUE ANALYSIS

In [17]:
# Printing all the columns with atleast one null value
zomatoCountrydt.columns[zomatoCountrydt.isna().any()].tolist()

['Cuisines']

From above:
#The dataset has 9551 rows and 22 columns
#There are 3 columns with float, 5 columns with int, 14 columns that are categorical
#There is one column 'Cuisines' that has missing values

# Exploratory Data Analysis (EDA)

Pandas Profiling

In [23]:
import pandas_profiling                             # Automatic EDA
import sweetviz as sv                               # Importing sweetviz for some sweet visualizations
import seaborn as sns  

ImportError: cannot import name 'to_html' from 'pandas_profiling.report' (C:\Users\satya\anaconda3\lib\site-packages\pandas_profiling\report\__init__.py)

In [24]:
# Making a copy of the dataset as df
dt = zomatoCountrydt.copy(deep = True)

In [25]:
# Generating the profile report and feeding it into a variable
# zomatoPandasProfile = ProfileReport(df, title = 'Zomato Data: Pandas Profiling Report', html = {'style':{'full_width':True}})

In [26]:
# Showcasing the Pandas Profiling Report for the Zomato Dataset
# zomatoPandasProfile

In [27]:
# We shall use the SweetViz analyze() function to generate a shareable .html file to analyze the dataset
analyze_report = sv.analyze(dt)
# analyze_report.show_html('zomatoSweetViz.html', open_browser=False)

NameError: name 'sv' is not defined

In [22]:
# Displaying the html file in the Google Colab notebook
# IPython.display.HTML('zomatoSweetViz.html')

Data Preparation

In [4]:
# Plotting the features of the dataframe
# plot(dt)

In [ ]:
------------------------ I tried many times to update my kernel and also tried to instal my notebook,
                      But no use, some libraries are not installing - withour that i cant move ahead in the program.
                      I tried my level best - and wil try again

In [ ]:
REMARKS - I tried a lot to manage the things ,
but i am unable to install some python libraries and import some sweetviz libraries. and also others.
# if i postively get all my errors done, further i will surely resubmit the assignment again.
Kinldy excuse for not geeting the results.